### PR TITLE
Add better wildcard file search support

### PR DESCRIPTION
**Pull request by david-poindexter**
Add better wildcard search support
- Style correction for `Item Name` sort dropdown
- Corrected wildcard regex logic for SearchFiles

### CODE REVIEW

File: DNN Platform/Modules/ResourceManager/Components/SearchController.cs [hunk 1]
```diff
@@ -34,26 +34,14 @@ public IList<IFileInfo> SearchFolderContent(int moduleId, IFolderInfo folder, bo
 
             search = (search ?? string.Empty).Trim();
 
-            // Lucene does not support wildcards as the beginning of the search
```
Comment (bdukes) [inline DNN Platform/Modules/ResourceManager/Components/SearchController.cs:37]: Did something change with Lucene's support for wildcards?  What's the behavior if you search with a wildcard at the beginning of your search?

File: DNN Platform/Library/Services/FileSystem/FolderManager.cs [hunk 2]
```diff
@@ -1845,11 +1845,11 @@ private static Regex WildcardToRegex(string pattern)
         {
             if (!pattern.Contains("*") && !pattern.Contains("?"))
             {
-                pattern = "^" + pattern + ".*$";
+                pattern = ".*" + Regex.Escape(pattern) + ".*";
```
[the remark this answers is not in the document]
Comment (mitchelsellers) [reply DNN Platform/Library/Services/FileSystem/FolderManager.cs:1848]: I too am curious on this one, the "old way" was requires full string match, starting with the pattern at the beginning.  For example

TestF as an input

Would match TestF, TestFile, TestFilters, TestF Test, etc

This new pattern would match those, but also would match

MyTestF, etc.

Is this a desired behavior?

**Comment by sleupold** [comment]
I agree, left search is most common and I don't see a need for supporting search in the middle of a word.

File: DNN Platform/Library/Services/FileSystem/FolderManager.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (david-poindexter) [reply DNN Platform/Library/Services/FileSystem/FolderManager.cs:1848]: > I too am curious on this one, the "old way" was requires full string match, starting with the pattern at the beginning.  For example
> 
> 
> 
> TestF as an input
> 
> 
> 
> Would match TestF, TestFile, TestFilters, TestF Test, etc
> 
> 
> 
> This new pattern would match those, but also would match
> 
> 
> 
> MyTestF, etc.
> 
> 
> 
> Is this a desired behavior?
> 
> 

Yes, this is the desired behavior.  It is a much more powerful wildcard search.

[the remark this answers is not in the document]
Comment (david-poindexter) [reply DNN Platform/Library/Services/FileSystem/FolderManager.cs:1848]: > If you don't have the start and end tokens, you wouldn't need the `.*` pattern, either, right?

I believe you would in order to achieve the "contains" search results.

File: DNN Platform/Modules/ResourceManager/Components/SearchController.cs (hunk 1 above)
Comment (david-poindexter) [reply DNN Platform/Modules/ResourceManager/Components/SearchController.cs:37]: @bdukes I couldn't for the life of me figure out why this was even there.  It works as expected without this unnecessary startsWith trimming.

**Comment by david-poindexter** [comment]
Just to be clear, this PR is in support of the new Resource Manager.  I should have made that clearer in my initial comments.  Notice the PR is against the feature branch. ;-)

File: DNN Platform/Library/Services/FileSystem/FolderManager.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (bdukes) [reply DNN Platform/Library/Services/FileSystem/FolderManager.cs:1848]: Regular Expression matching is a "contains" match by default, here's an example:
LINQPad screenshot

[the remark this answers is not in the document]
Comment (david-poindexter) [reply DNN Platform/Library/Services/FileSystem/FolderManager.cs:1848]: It looks like this was merged already, so I'll do another PR to change this. Thanks @bdukes

**Pull request merged**